cmd/harmony/main.go

```diff
@@ -157,37 +194,85 @@ func raiseFdLimits() error {
 	return nil
 }
 
-func getHarmonyConfig(cmd *cobra.Command) (harmonyConfig, error) {
+func promptForConfigUpgrade() bool {
+	if !terminal.IsTerminal(syscall.Stdin) {
+		return false
+	}
+	for {
+		fmt.Fprint(os.Stderr, "Would you like to upgrade your config to the latest version? (y/n): ")
+		var input string
+		fmt.Scanln(&input)
+		switch strings.ToLower(input) {
+		case "y":
+			return true
+		case "n":
+			return false
+		default:
+			fmt.Fprint(os.Stderr, "Incorrect input. ")
+		}
+	}
+}
+
+func runUpgradeConfig(cmd *cobra.Command, args []string) {
 	var (
-		config harmonyConfig
-		err    error
+		cfg        config.HarmonyConfig
+		configFile string
+		err        error
+	)
+	if cli.IsFlagChanged(cmd, configFlag) {
+		configFile = cli.GetStringFlagValue(cmd, configFlag)
+		cfg, err = config.LoadHarmonyConfig(configFile)
+	} else {
+		fmt.Println("config file should be specified")
+		os.Exit(128)
+	}
+	if cfg.Version == config.TOMLConfigVersion {
+		// nothing to do
+		return
+	}
+	oldFileName, err := config.BackupAndUpgradeConfigToTheLatestVersion(cfg, configFile)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(128)
+	}
+	fmt.Printf("previous config was stored at %s\n", oldFileName)
+}
+
+func getHarmonyConfig(cmd *cobra.Command) (config.HarmonyConfig, error) {
+	var (
+		cfg config.HarmonyConfig
+		err error
 	)
 	if cli.IsFlagChanged(cmd, configFlag) {
 		configFile := cli.GetStringFlagValue(cmd, configFlag)
-		config, err = loadHarmonyConfig(configFile)
+		cfg, err = config.LoadHarmonyConfig(configFile)
 	} else {
 		nt := getNetworkType(cmd)
-		config = getDefaultHmyConfigCopy(nt)
+		cfg = config.GetDefaultHmyConfigCopy(nt)
 	}
 	if err != nil {
-		return harmonyConfig{}, err
+		return config.HarmonyConfig{}, err
 	}
-	if config.Version != defaultConfig.Version {
+	if cfg.Version != config.DefaultConfig.Version {
```

Comment:
Refactor to `config.IsLatestVersion(versionStr string)`? Also shall we consider the config version larger than current version?

Reply:
great idea! What if I add IsLatestVersion() method on the HarmonyConfig struct itself?